### DISCUSSION TITLE
True embedded mode?

### DISCUSSION THREAD

**Discussion by lucian-cm** [ArcadeData/arcadedb#866] [Q&A] [answered]
I believe the multi-model database concept that ArcadeDB supports is a very powerful functionality. 

There are use cases where ArcadeDB can be used as a client-side database (similar to how SQLite or H2
are embedded) but to leverage ArcadeDB's graph/document/multi-model which are not
present in the above mentioned databases (e.g. Desktop applications, non server applications that have graph/document/multi-model needs aso).
 
In such use cases, there is usually no need to be able to access the database from outside (as all data access patterns come from the same process, namely the embedding application process)
, so having a server application would increase the attack surface and raise additional security questions, without leveraging 
the benefit of having remote clients connecting to the database.

For such use cases, it would be nice to be able to leverage a `true embedded mode (server-less mode)` instead of an 
`embedded server mode`, which would allow such embedding applications to be more lightweight 
(no need to have even a lightweight server such as Undertow on the classpath) and have better security (application does not expose any communication endpoint nor it is listening on any port). 
Basically in this true embedded mode, ArcadeDB would not start any server component like Undertow. 
Also in most of these usecases, there is no need for data visualization tools like ArcadeDB Studio to connect to the database
as the embedding app has its own way to show the data, for the cases when this data is in fact user facing. 

@ ArcadeDB maintainers and its community: Do you feel that supporting this new embedded mode is something worth considering?
Any comments or suggestions are welcome.

**Comment by kran** [comment, marked as answer]
I think there is a "real embedded mode" for arcadedb: https://docs.arcadedb.com/#Java-API

**Reply by lvca** [reply]
As @kran reported, the truly embedded mode uses your database inside your JVM with the Java API. Super fast, zero security issues, +100!

**Reply by lucian-cm** [reply]
That is great 👍 
I missed that section from the documentation, thanks @kran, @lvca for your quick responses.
